lib/node_modules/@stdlib/repl/lib/tokenizer.js

```diff
@@ -0,0 +1,359 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* eslint-disable max-lines-per-function */
+
+'use strict';
+
+// MODULES //
+
+var parse = require( 'acorn-loose' ).parse;
+var walk = require( 'acorn-walk' );
+var linkedList = require( '@stdlib/utils/linked-list' );
+var contains = require('@stdlib/array/base/assert/contains');
+var resolveLocalScopes = require( './resolve_local_scopes.js' );
+var resolveLocalScope = require( './resolve_local_scope.js' );
+var commands = require( './commands.js' );
+
+
+// VARIABLES //
+
+var COMMANDS = commands();
+var isControlKeyword = contains.factory( [ 'if', 'else', 'switch', 'case', 'catch', 'finally', 'try', 'return', 'break', 'continue' ] );
+var isSpecialIdentifier = contains.factory( [ 'this', 'super' ] );
+var isReservedLiteral = contains.factory( [ 'null', 'true', 'false' ] );
+var isUnrecognizedKeyword = contains.factory( [ 'async', 'await', 'let' ] );
+var isStringTokenType = contains.factory( [ 'string', 'template' ] );
+var isLiteralType = contains.factory( [ 'string', 'boolean', 'number' ] );
+var isReservedName = contains.factory( [ 'undefined', 'async', 'await', 'let' ] );
+var RE_PUNCTUATION = /(\{|\}|\[|\]|\(|\)|,|;|:|\.|\?|\?\.|=>|\.\.\.|`|\${)/;
```

Comment:
We could consider moving this into its own `@stdlib/regexp/*` package down the line.